faraday/proxy.py

```diff
@@ -393,6 +428,208 @@ def testdb_read_worker():
         row = cursor.fetchone()
 
 
+def acceptConnection(server, faraday):
+    conn, addr = server.accept()
+    logger.info("Got connection from {0} on {1}".format(addr, faraday))
+    return conn, addr
+
+
+def closeConnection(conn, addr, faraday):
+    # close the connection
+    logger.info("Closing connection with {0} on {1}".format(addr, faraday))
+    try:
+        conn.close()
+    except socket.error as e:
+        logger.error(e)
+
+
+def extractBytes(data, dataBuffer, unit):
+    # data is a BASE64 string with unknown length, iterate and separete
+    #data = data.decode('base64', 'strict')
+
+    for byte in data:
+        try:
+            byte = struct.unpack("c", byte)[0]
+            dataBuffer[unit].append(byte)
+
+        except:
+            logger.debug("Creating dataBuffer")
+            dataBuffer[unit] = deque([])
+            dataBuffer[unit].append(byte)
+
+
+def receiveData(conn, addr, dataBuffer, unit):
+    while True:
+        try:
+            data = conn.recv(4096)
+        except socket.error as e:
+            logger.error(e)
+            closeConnection(conn, addr, unit)
+            break
+
+        if not data:
+            closeConnection(conn, addr, unit)
+            break
+
+        # Expect BASE64 so decode it
+        extractBytes(data, dataBuffer, unit)
+
+
+def sendData(conn, addr, getDicts, unit, payloadSize):
+    while True:
+        # not sure why I need a delay... otherwise socket closes
+        time.sleep(0.05)
+        dataQueue = []
+        try:
+            dataQueue = getDicts[unit][1]
+        except KeyError as e:
+            # Simply haven't ever received data so break
+            pass
+
+        if len(dataQueue) <= 0:
+            conn.sendall("No Data! Goodbye.")
+            closeConnection(conn, addr, unit)
+            break
+
+        try:
+            # pop off a data entry from the left of getDicts
+            temp = dataQueue.popleft()
+
+        except IndexError as e:
+            # Empty queue
+            logger.error(e)
+
+        try:
+            data = temp['data'].decode('base64', 'strict')
+
+        except StandardError as e:
+            logger.error(e)
+
+        # unpack frames and retrieve data originally sent to socket
+        try:
+            dataList = struct.unpack("BB121s", data)
+            dataList2 = struct.unpack("B{0}s".format(payloadSize), dataList[2][:payloadSize + 1])
+            socketData = dataList2[1][:dataList2[0]]
```

Comment:
Add comment to describe that dataList2[0] is the packed length in bytes of the payload data.